app/src/main/java/com/codesoom/assignment/domain/User.java

```diff
@@ -37,6 +39,12 @@ public void destroy() {
         deleted = true;
     }
 
+    /**
+     * 패스워드가 일치하다면 true, 일치하지 않거나 삭제된 유저라면 false를 리턴합니다.
+     *
+     * @param password 비교할 패스워드
+     * @return 패스워드가 일치하면 true, 일치하지 않거나 삭제되었다면 false
+     */
```

Comment:
핵심을 보셔야 해요. 여기에서 핵심은 패스워드가 일치하는지가 아닙니다. 그렇기 때문에 `패스워드가 일치하지 않거나 삭제된 유저라면...` 처럼 메소드 구현에 의존하는 주석이 나온 거라 할 수 있어요. 한편 이 메소드가 정말로 여기에 있어야 하는지에 대해서는 열심히 생각해 봐야 하는 문제이기도 합니다. 일단 이 메소드는 그냥 두고... 좀 더 기다려서 때가 되면 옮겨 봅시다.

```suggestion
    /**
     * 인증에 성공하면 true, 인증에 실패하면 false를 리턴합니다.
     *
     * @param password 비교할 패스워드
     * @return 인증에 성공하면 true, 인증에 실패하면 false
     */
```

Reply:
반영하였습니다.
말씀하신대로 이 메서드가 여기에 있는 것은 적절한지도 원래 고민하던 부분이였는데, 아직 조금 더 고민이 필요한 것 같습니다.

Reply:
좋습니다. 고민을 더 해보세요. 이런 고민은 제가 대신해드리면 실력이 늘지 않겠죠. 열심히 생각해서 생각한 것들을 설명해 보세요.

Reply:
회원이 자기 자신을 인증한다라는 것이 조금 이상하다는 생각이 문뜩 들었습니다.
삭제되었는지 확인하는 메서드, 패스워드가 일치하는지 검증하는 메서드 이렇게 2개로 분리하여
이 메서드를 호출하는 쪽에서 둘 중 하나라도 해당된다면 에러를 던지게 하는걸로 바꿀 수 있을 것 같습니다.

```java
if (user.isDeleted() || !user.isMatchPassword()) {
    // 에러를 던진다
}
```

Reply:
해볼만한 생각입니다. 여기에서 더 생각해 봐야 할 것은 두 가지가 있어요.

1. 인증은 누구의 책임인가? 인증이 누구의 책임인지를 명확히 할 수 있다면 인증 코드가 어디로 가야 할지도 명확해질 겁니다.
2. "회원이 자기 자신을 인증하는 행위"가 이상하게 느껴지다면 인증을 전담하는 객체를 만들어야 할 때일지도 모릅니다. 한편, 이것이 이상하게 느껴지는 것은 현실에서의 이야기이기도 해요. 만약 회원 클래스가 클래스라고 생각하지 말고, 독립된 컴퓨터나 서비스라고 생각해 봅시다. "회원 서비스"라면 회원 정보를 전송해서 인증되는지를 응답해달라고 하는 것은 크게 이상하게 느껴지지 않겠죠.